src/classes/BDI_DataImport_BATCH.cls

```diff
@@ -465,8 +504,15 @@ public with sharing class BDI_DataImport_BATCH implements Database.Batchable<sOb
                 setEmail.add(di.Contact1_Work_Email__c);            
             if (di.Contact1_Alternate_Email__c != null)
                 setEmail.add(di.Contact1_Alternate_Email__c);            
-            if (isCustomIDInContactMatchRules && di.get('Contact1_' + diSettings.Contact_Custom_Unique_ID__c + strTestContactUniqueIdSuffix) != null)
-                setCustomID.add(string.valueOf(di.get('Contact1_' + diSettings.Contact_Custom_Unique_ID__c + strTestContactUniqueIdSuffix)));
+            if (isCustomIDInContactMatchRules) { 
+                object val = di.get('Contact1_' + diSettings.Contact_Custom_Unique_ID__c + strTestContactUniqueIdSuffix);
```

Comment:
For my understanding, could you explain why are we adding __c only if we are running tests?